linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/bash/shared.sh

```diff
@@ -1,3 +1,20 @@
 # platform = multi_platform_wrlinux,multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv,multi_platform_sle
-sed --follow-symlinks -i 's/\<nullok\>//g' /etc/pam.d/system-auth
-sed --follow-symlinks -i 's/\<nullok\>//g' /etc/pam.d/password-auth
+SYSTEM_AUTH="/etc/pam.d/system-auth"
+PASSWORD_AUTH="/etc/pam.d/password-auth"
+
+if [ -f /usr/bin/authselect ]; then
+    if [ $(authselect enable-feature without-nullok) ]; then
```

Comment:
I wonder in what case this `enable-feature` would fail where the second `enable-feature`, in the `else` would not.

Reply:
The first case will result success (true) if the profile is integer. In this case, we don't need to force the profile selection. In case this command fails, it means the authselect profile is not integer. For this case, we reactivate it, preserving the features but removing manual modifications which caused the integrity check to fail.

In Ansible, the logic is the same, but instead or directly trying to apply the change, the check option is used in order to make the Ansible tasks easier to be executed by conditionals.